starknet_py/net/models/transaction.py

```diff
@@ -76,13 +86,54 @@ class AccountTransaction(Transaction, ABC):
 TypeAccountTransaction = TypeVar("TypeAccountTransaction", bound=AccountTransaction)
 
 
+@dataclass(frozen=True)
+class DeclareV3(AccountTransaction):
```

Comment:
Could this dataclass be manually instantiated by a user? If it is, we should probably limit the values that could be passed to a constructor.

Reply:
Added `field(init=False)` to fields that shouldn't be instantiated by a user. I have also created internal class `_AccountTransactionV3` with a helper function `get_common_fields()`